plasma_framework/contracts/src/utils/PosLib.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity 0.5.11;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @dev UTXO position = (blockNumber * BLOCK_OFFSET + txIndex * TX_OFFSET + outputIndex).
+ * TX position = (blockNumber * BLOCK_OFFSET + txIndex * TX_OFFSET)
+ */
+library PosLib {
+    using SafeMath for uint256;
+
+    struct Position {
+        uint256 blockNum;
+        uint256 txIndex;
+        uint16 outputIndex;
+    }
+
+    uint256 constant internal BLOCK_OFFSET = 1000000000;
+    uint256 constant internal TX_OFFSET = 10000;
+
+    /**
+     * @notice Returns the UTXO position for a given transaction position and output index
+     * @param txPos Transaction position - utxo position of zero index output
+     * @param outputIndex Index of the output
+     * @return utxo position
+     */
+    function buildPositionFromTxPosAndOutputIndex(uint256 txPos, uint16 outputIndex)
```

Comment:
I am actually thinking to change the interface of spending condition a little bit. Instead of 
```
function verify(
        bytes calldata inputTxBytes,
        uint16 outputIndex,
        uint256 inputTxPos,
        bytes calldata spendingTxBytes,
        uint16 inputIndex,
        bytes calldata signature
    )
```

merges the `outputIndex` and `inputTxPos` to `utxoPos` or `outputPos`.
so sth like:
```
function verify(
        bytes calldata inputTxBytes,
        PosLib.Position calldata utxoPos,
        bytes calldata spendingTxBytes,
        uint16 inputIndex,
        bytes calldata signature
    )
```

It was originally separated to make it clearer that the `txPos` might potentially be 0 if we are doing MoreVP 2.0, so where you can chain in-flight txs. However, I think that is far away and it is still possible to support that by giving the `utxoPos` with outputIndex only.

If doing so, it removes the need of this `buildPositionFromTxPosAndOutputIndex` function. Also, it removes one use case of `encodePackedTxPos` function.

Reply:
Done, but because of unspecified compilation error (compilation freeze) I couldn't use `PosLib.Position` struct in the interface API, I needed to use `uint256` for `utxoPos`.